internals/resource-managers/nfspvc.go

```diff
@@ -0,0 +1,107 @@
+package resourceprepares
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	cappv1alpha1 "github.com/dana-team/container-app-operator/api/v1alpha1"
+	rclient "github.com/dana-team/container-app-operator/internals/wrappers"
+	nfspvcv1alpha1 "github.com/dana-team/nfspvc-operator/api/v1alpha1"
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/record"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type NFSPVCManager struct {
+	Ctx           context.Context
+	K8sclient     client.Client
+	Log           logr.Logger
+	EventRecorder record.EventRecorder
+}
+
+// prepareResource prepares the NfsPvc resource based on the Capp object.
+func (n NFSPVCManager) prepareResource(capp cappv1alpha1.Capp) []nfspvcv1alpha1.NfsPvc {
+	nfsPvcs := []nfspvcv1alpha1.NfsPvc{}
+	for _, nfsVolume := range capp.Spec.VolumesSpec.NFSVolumes {
+		nfsPvc := nfspvcv1alpha1.NfsPvc{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      nfsVolume.Name,
+				Namespace: capp.Namespace,
+				Labels: map[string]string{
+					CappResourceKey: capp.Name,
+				},
+			},
+			Spec: nfspvcv1alpha1.NfsPvcSpec{
+				Server: nfsVolume.Server,
+				Path:   nfsVolume.Path,
+				AccessModes: []corev1.PersistentVolumeAccessMode{
+					corev1.ReadWriteMany,
+				},
```

Comment:
Probably should add a comment here similar to what you wrote on the PR to explain why this is the only relevant AccessMode for future readers